packages/network-controller/src/rpc-service/rpc-service-chain.test.ts

```diff
@@ -645,22 +749,46 @@ describe('RpcServiceChain', () => {
       };
       // Retry the first endpoint until max retries is hit.
       await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
-        'Gateway timeout',
+        expect.objectContaining({
+          code: -32002,
+          message: 'RPC endpoint not found or unavailable.',
+          data: {
+            httpStatus: 503,
+          },
+        }),
       );
       // Retry the first endpoint again, until max retries is hit.
       await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
-        'Gateway timeout',
+        expect.objectContaining({
+          code: -32002,
+          message: 'RPC endpoint not found or unavailable.',
+          data: {
+            httpStatus: 503,
+          },
+        }),
       );
       // Retry the first endpoint for a third time, until max retries is hit.
       // The circuit will break on the last time, and the second endpoint will
       // be retried, until max retries is hit.
       await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
-        'Gateway timeout',
+        expect.objectContaining({
+          code: -32002,
+          message: 'RPC endpoint not found or unavailable.',
+          data: {
+            httpStatus: 503,
+          },
+        }),
       );
       // Try the first endpoint, see that the circuit is broken, and retry the
       // second endpoint, until max retries is hit.
       await expect(rpcServiceChain.request(jsonRpcRequest)).rejects.toThrow(
-        'Gateway timeout',
+        expect.objectContaining({
+          code: -32002,
+          message: 'RPC endpoint not found or unavailable.',
```

Comment:
i go back and forth on how clean/DRY tests need to be. This seems to be repeated quite a bit, but up to you if you feel it warrants being moved into a constant or if you feel that the tests are still readable as is